src/core/public/workspace/workspaces_service.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { BehaviorSubject, combineLatest } from 'rxjs';
+import { isEqual } from 'lodash';
+
+import { CoreService, WorkspaceAttribute } from '../../types';
+
+type WorkspaceObject = WorkspaceAttribute & { readonly?: boolean };
+
+interface WorkspaceObservables {
+  /**
+   * Indicates the current activated workspace id, the value should be changed every time
+   * when switching to a different workspace
+   */
+  currentWorkspaceId$: BehaviorSubject<string>;
+
+  /**
+   * The workspace that is derived from `currentWorkspaceId` and `workspaceList`, if
+   * `currentWorkspaceId` cannot be found from `workspaceList`, it will return an error
+   *
+   * This value MUST NOT set manually from outside of WorkspacesService
+   */
+  currentWorkspace$: BehaviorSubject<WorkspaceObject | null>;
```

Comment:
doesn't the `currentWorkspace` object include workspace id that we need to have another observable for workspace id?

Reply:
Right, ideally having `currentWorkspaceId` and `workspaceList` is good enough, because `currentWorkspace` can be derived from `currentWorkspaceId` and `workspaceList`. 

However, in the actual code, the places that SET the current workspace normally only have the `id` and the places that GET the current workspace normally will need the entire workspace object.

The `currentWorkspace$` is more like a `helper`, its computed from `currentWorkspaceId$` and `workspaceList$` once they changed.